### PR TITLE
Fix oneOf/anyOf with extra properties

### PR DESCRIPTION
**Pull request by Porges**

### CODE REVIEW

**Review by theunrepentantgeek — APPROVED** [review]
LGTM. :shipit:

**Comment by Porges** [comment]
Wait this is completely wrong. One moment…

**Review by theunrepentantgeek — APPROVED** [review]

File: hack/generator/pkg/jsonast/jsonast.go [hunk 1]
```diff
@@ -628,7 +628,19 @@ func generateOneOfUnionType(ctx context.Context, subschemas []Schema, scanner *S
 		}
 	}
 
-	return astmodel.MakeOneOfType(types...), nil
+	result := astmodel.MakeOneOfType(types...)
+
+	// if the node that contains the oneOf(/anyOf) defines other properties, create an object type with them inside to merge
```
Comment (matthchr) [inline hack/generator/pkg/jsonast/jsonast.go:633]: You could add a golden file test to ensure this does the right thing?

We have some already in the OneOf folder I think.

**Review by matthchr — APPROVED** [review]

**Comment by matthchr** [comment]
LGTM!

**Pull request merged**